compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java

```diff
@@ -238,6 +240,15 @@ private void loadDefaultValue(MethodVisitor mv, BType bType) {
             case TypeTags.STREAM:
                 mv.visitInsn(ACONST_NULL);
                 break;
+            case TypeTags.DECIMAL:
+                mv.visitTypeInsn(NEW, DECIMAL_VALUE);
+                mv.visitInsn(DUP);
+                mv.visitInsn(DCONST_0);
+                mv.visitMethodInsn(INVOKESTATIC, BIG_DECIMAL, VALUE_OF_METHOD, String.format("(D)L%s;", BIG_DECIMAL),
+                        false);
+                mv.visitMethodInsn(INVOKESPECIAL, DECIMAL_VALUE, JVM_INIT_METHOD, String.format("(L%s;)V", BIG_DECIMAL),
+                        false);
```

Comment:
why do we need a `NEW` instruction. isn't following enough
```suggestion
                mv.visitInsn(DCONST_0);
                mv.visitMethodInsn(INVOKESTATIC, BIG_DECIMAL, VALUE_OF_METHOD, String.format("(D)L%s;", BIG_DECIMAL),
                        false);
```

Reply:
I've used ```new DecimalValue(BigDecimal.valueOf(0.0));``` code segment to generate the decimal value. So I'm needed to add `DecimalValue` path. Without those lines, program crashed. So we need those lines.